docs/_design/map-controller.md

```diff
@@ -0,0 +1,399 @@
+between some object (the parent node) and the objects that are directly under it
+in an ownership tree (child nodes).
+This is analogous to the [Composite pattern][] in that it lets you manage a
+group of child objects as if were one object (by manipulating only the parent
+object).
+
+Similarly, DecoratorController lets you add new child nodes to a parent node
+that already has some other behavior.
+This is analogous to the [Decorator pattern][] in that it lets you dynamically
+wrap new behavior around select instances of an existing object type without
+having to create a new type.
+
+[Composite pattern]: https://en.wikipedia.org/wiki/Composite_pattern
+[Decorator pattern]: https://en.wikipedia.org/wiki/Decorator_pattern
+
+## Problem Statement
+
+The problem that MapController addresses is that neither CompositeController nor
+DecoratorController allow you to make decisions based on objects that
+aren't owned by the particular parent object being processed.
+That's because in the absence of a parent-child relationship, there are
+arbitrarily many ways you could pick what other objects you want to look at.
+
+To avoid having to send every object in a given resource (e.g. every Pod)
+on every hook invocation, there must be some way to tell Metacontroller which
+objects you need to see (that you don't own) to compute your desired state.
+Rather than try to embed various options for declaring these relationships
+(object name? label selector? field selector?) into each existing Metacontroller
+API, the goal of MapController is to provide a solution that's orthogonal to the
+existing APIs.
+
+In other words, we attempt to separate the task of looking at non-owned objects
+(MapController) from the task of defining objects that are composed of other
+objects (CompositeController) so that users can mix and match these APIs
+(and future APIs) as needed without being limited to the precise scenarios we're
+able to anticipate.
+
+## Proposed Solution
+
+MapController lets you define a collection of objects owned by a parent object,
+where each child object is generated by some mapping from a non-owned object.
+This is analogous to the general concept of a [map function][] in that it calls
+your hook for each object in some input list (of non-owned objects), and creates
+an output list (of child objects) containing the results of each call.
+
+A single `sync` pass for a MapController roughly resembles this pseudocode:
+
+```
+def sync_map_controller():
+  input_list = get_matching_objects(input_resource, input_selector)
+  output_list = list()
+
+  foreach input_object in input_list:
+    output_list.append(map_hook(input_object))
+
+  reconcile_objects(output_list)
+```
+
+where `map_hook()` is the only code that the MapController user writes,
+as a lambda hook.
+
+In general, MapController addresses use cases that can be described as,
+"For every matching X that already exists, I want to create a Y according to the
```

Comment:
To confirm, we can create multiple Ys right?
And in this case, does the parent object = X = non-owned object, and Y = child object?

Reply:
Good point. I've updated the wording to clarify there can be any number of Y objects for each X. Thanks!